go/billing/tests/test_tasks.py

```diff
@@ -449,12 +524,224 @@ def test_generate_monthly_statement_sessions_none_markup(self):
             self.account.id, *this_month())
 
         [item] = get_line_items(statement).filter(
-            description='Sessions')
+            description='Sessions (billed per session) - provider1')
 
         self.assertEqual(item.credits, None)
         self.assertEqual(item.unit_cost, 100)
         self.assertEqual(item.cost, 100)
 
+    @mock.patch('go.billing.settings.PROVIDERS', {
+        'provider1': 'Provider 1',
+        'provider2': 'Provider 2',
+    })
+    def test_generate_monthly_statement_providers_inbound(self):
+        mk_transaction(
+            self.account,
+            provider='provider1',
+            message_cost=100,
+            markup_percent=10.0,
+            message_direction=MessageCost.DIRECTION_INBOUND)
+
+        mk_transaction(
+            self.account,
+            provider='provider2',
+            message_cost=100,
+            markup_percent=10.0,
+            message_direction=MessageCost.DIRECTION_INBOUND)
+
+        mk_transaction(
+            self.account,
+            provider='provider1',
+            message_cost=100,
+            markup_percent=10.0,
+            message_direction=MessageCost.DIRECTION_INBOUND)
+
+        statement = tasks.generate_monthly_statement(
+            self.account.id, *this_month())
+
+        [item1, item2] = get_line_items(statement).filter(
+            description__startswith='Messages received')
+
+        self.assertEqual(item1.description, 'Messages received - Provider 1')
+        self.assertEqual(item2.description, 'Messages received - Provider 2')
```

Comment:
I'm not sure if this is relevant (I'm not sure how we want the statement to be grouped), but would it be useful to test that the providers are being  grouped correctly, ie. Make sure that the quantity is 2 for `provider1` and 1 for `provider2`.